libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/HttpHelper.cs

```diff
@@ -28,19 +28,26 @@ internal static class HttpHelper
 
         public static Activity ReadRequest(HttpRequest request)
         {
-            if (request == null)
+            try
             {
-                throw new ArgumentNullException(nameof(request));
-            }
+                if (request == null)
+                {
+                    throw new ArgumentNullException(nameof(request));
+                }
+
+                var activity = default(Activity);
 
-            var activity = default(Activity);
+                using (var bodyReader = new JsonTextReader(new StreamReader(request.Body, Encoding.UTF8)))
+                {
+                    activity = BotMessageSerializer.Deserialize<Activity>(bodyReader);
+                }
 
-            using (var bodyReader = new JsonTextReader(new StreamReader(request.Body, Encoding.UTF8)))
+                return activity;
+            }
+            catch (JsonException)
             {
-                activity = BotMessageSerializer.Deserialize<Activity>(bodyReader);
+                return null;
```

Comment:
This should log, using our infrastructure so it ends up in the right place. I can see that being important to debugging sessions by the Supportability team.

Reply:
This will be logged at the HTTP level. This change uses the correct HTTP error code.